### PR TITLE
Feature/frontend/messages scroll to bottom/karanmeet

### PR DESCRIPTION
**Pull request by karanmeetkhatra10**

### CODE REVIEW

File: Online-Messaging-App/src/app/home/chatbox/chatbox.component.ts [hunk 1]
```diff
@@ -32,19 +35,30 @@ export class ChatboxComponent implements OnInit {
...
+            this.scrollToBottom();
         });
+
+    }
+
+    ngAfterViewChecked() {
+        this.scrollToBottom();
+    }
+
+    ngAfterViewInit(): void {
```
Comment (dBaiko) [inline Online-Messaging-App/src/app/home/chatbox/chatbox.component.ts]: Remove this

File: Online-Messaging-App/src/app/home/chatbox/chatbox.component.ts [hunk 2]
```diff
@@ -32,19 +35,30 @@ export class ChatboxComponent implements OnInit {
...
     @Input()
     set channelId(value: any) {
         this._channelId = value;
         this.getMessages(this._channelId);
+        this.isNearBottom = false;
     }
 
     ngOnInit(): void {
         this.messagerService.subscribeToSocket().subscribe((data) => {
             if (data.channelId == this.channelId) this.chatMessages.push(data);
```
Comment (dBaiko) [inline Online-Messaging-App/src/app/home/chatbox/chatbox.component.ts]: please fix this if statement, please add proper {}

File: Online-Messaging-App/src/app/home/chatbox/chatbox.component.ts [hunk 3]
```diff
@@ -68,10 +82,38 @@ export class ChatboxComponent implements OnInit {
...
+        console.log(this.atBottom);
+        if (this.atBottom) {
+            this.isNearBottom = false;
+        } else {
+            this.isNearBottom = true;
+        }
+    }
+
+    private scrollToBottom(): void {
+        try {
```
Comment (dBaiko) [inline Online-Messaging-App/src/app/home/chatbox/chatbox.component.ts:100]: Remove unneccisary try

File: Online-Messaging-App/src/app/home/chatbox/chatbox.component.ts [hunk 4]
```diff
@@ -68,10 +82,38 @@ export class ChatboxComponent implements OnInit {
...
+
+            if (this.isNearBottom) {
+                return;
+            }
+            try {
+                this.scrollContainer.nativeElement.scrollTop = this.scrollContainer.nativeElement.scrollHeight;
+                this.isNearBottom = false;
+            } catch (err) {
+            }
+            // this.scrollContainer.nativeElement.scrollTop = this.scrollContainer.nativeElement.scrollHeight;
```
Comment (dBaiko) [inline Online-Messaging-App/src/app/home/chatbox/chatbox.component.ts]: remove

File: Online-Messaging-Backend/package.json [hunk 5]
```diff
@@ -22,6 +22,7 @@
     "express": "^4.17.1",
     "helmet-csp": "^2.9.4",
     "nodemon": "^1.19.4",
+    "scroll-js": "^3.1.1",
```
Comment (dBaiko) [inline Online-Messaging-Backend/package.json]: Remove this, unused

**Review by dBaiko — CHANGES_REQUESTED** [review]

File: Online-Messaging-Backend/package.json (hunk 5 above)
Comment (karanmeetkhatra10) [reply Online-Messaging-Backend/package.json]: Done

File: Online-Messaging-App/src/app/home/chatbox/chatbox.component.ts (hunk 4 above)
Comment (karanmeetkhatra10) [reply Online-Messaging-App/src/app/home/chatbox/chatbox.component.ts]: Done

File: Online-Messaging-App/src/app/home/chatbox/chatbox.component.ts (hunk 3 above)
Comment (karanmeetkhatra10) [reply Online-Messaging-App/src/app/home/chatbox/chatbox.component.ts:100]: Done

File: Online-Messaging-App/src/app/home/chatbox/chatbox.component.ts (hunk 2 above)
Comment (karanmeetkhatra10) [reply Online-Messaging-App/src/app/home/chatbox/chatbox.component.ts]: Done

File: Online-Messaging-App/src/app/home/chatbox/chatbox.component.ts (hunk 1 above)
Comment (karanmeetkhatra10) [reply Online-Messaging-App/src/app/home/chatbox/chatbox.component.ts]: Done

**Review by dBaiko — APPROVED** [review]

**Pull request merged**